redash/models/__init__.py

```diff
@@ -138,28 +191,40 @@ def get_by_id(cls, _id):
         return cls.query.filter(cls.id == _id).one()
 
     def delete(self):
+        # Delete the relevant metadata about a data source first.
+        tables = TableMetadata.query.filter(TableMetadata.data_source_id == self.id).options(load_only('id'))
+        ColumnMetadata.query.filter(
+            ColumnMetadata.table_id.in_(tables.subquery())
+        ).delete(synchronize_session=False)
```

Comment:
Won't cascaded delete be enough here?

Reply:
Yes it is! Thanks for pointing that out. Updated.